src/Helper/Encryption/AbstractEncryptionHelper.php

```diff
@@ -41,20 +41,21 @@ public function decrypt($key, $data)
      */
     protected function getIV()
     {
-        return str_repeat(self::NULL_CHAR, mcrypt_get_iv_size($this->getCipher(), $this->getMode()));
+        return str_repeat(self::NULL_CHAR, openssl_cipher_iv_length($this->getCipherMethod()));
     }
 
     /**
-     * Returns one of the MCRYPT_ciphername constants of the name of the algorithm as string.
+     * Returns one of the available cipher methods
+     * @see openssl_get_cipher_methods()
      *
      * @return string
      */
-    abstract protected function getCipher();
+    abstract protected function getCipherMethod();
 
     /**
-     * Returns one of the MCRYPT_MODE_modename constants of one of "ecb", "cbc", "cfb", "ofb", "nofb" or "stream".
+     * @see openssl_decrypt() and openssl_encrypt() parammeters
      *
-     * @return string
+     * @return int
      */
-    abstract protected function getMode();
+    abstract protected function getEncodingStringSettings();
```

Comment:
1) It's not settings but options according to the [documentation](http://php.net/openssl_decrypt).
2) Why the method name contain "String" if it operates with integer value?